src/app/core/layer.ts

```diff
@@ -0,0 +1,385 @@
+            this.addFeature(Feature.fromInterchangeV1(json[i]));
+        }
+    }
+
+    // TODO: Replace Params and remove static method
+    /**
+     * Converts the attributes of the object into a JSON format
+     * @returns {JSON} Returns a JSON format with the attributes of the object
+     * @memberof Layer
+     */
+    toJSON() {
+        const output = {};
+        output.name = this.name.toJSON();
+        output.color = this.color;
+        output.params = this.params.toJSON();
+        output.features = this.__featuresToJSON();
+        return output;
+    }
+
+    /**
+     * Converts the attributes of the object into Interchange format
+     * @returns {*} Returns a Interchange format with the attributes of the object
+     * @memberof Layer
+     */
+    toInterchangeV1() {
+        const output = {};
+        output.id = this.__id;
+        output.name = this.name;
+        output.type = this.type;
+        // TODO - Add group and unique name parameters to the system and do type checking
+        // against type and not name in the future
+        output.group = "0";
+        output.params = this.params.toJSON();
+        return output;
+    }
+
+    /**
+     * Generate the feature layer json that is neccissary for
+     * seriailizing the visual of the 3DuF designs
+     *
+     * @returns {*} json of the features
+     * @memberof Layer
+     */
+    toFeatureLayerJSON() {
+        const output = {};
+        output.name = this.name;
+        output.color = this.color;
+        output.params = this.params.toJSON();
+        output.features = this.__featuresInterchangeV1();
+        return output;
+    }
+
+    /**
+     * Load from a JSON format a new layer object
+     * @param {JSON} json JSON format
+     * @returns {Layer} Returns a new layer object
+     * @memberof Layer
+     */
+    static fromJSON(json) {
+        if (!Object.prototype.hasOwnProperty.call(json, "features")) {
+            throw new Error("JSON layer has no features!");
+        }
+        const newLayer = new Layer(json.params, json.name);
+        newLayer.__loadFeaturesFromJSON(json.features);
+        if (json.color) newLayer.color = json.color;
+        return newLayer;
+    }
+
+    /**
+     * Load from an Interchange format a new layer object
+     * @param {*} json
+     * @returns {Layer} Returns a new layer object
+     * @memberof Layer
+     */
+    static fromInterchangeV1(json) {
+        const newLayer = new Layer(json.params, json.name, json.type, json.group);
+        newLayer.__loadFeaturesFromInterchangeV1(json.features);
+        if (json.color) newLayer.color = json.color; // TODO: Figure out if this needs to change in the future
+        return newLayer;
+    }
+
+    /**
+     * Render in 2D the features
+     * @param {*} paperScope
+     * @returns {Feature} Returns the renders
+     * @memberof Layer
+     */
+    render2D(paperScope) {
+        //Why is paperscope here?
+        return this.__renderFeatures2D();
+    }
```

Comment:
'paperScope' is never used. It's appearance as an argument results in an error in 'device'. Should I delete it or is there some functionality missing?

Reply:
Leave it there, we will deal with it later